### PR TITLE
Server timing header

### PR DESCRIPTION
**Pull request by EddeCCC**
Add new function getTraceContext() in InspectitContextImpl to extract the trace context and use it inside actions

- - -
This change is [Reviewable](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617)

### CODE REVIEW

**Review by TitusLabs — CHANGES_REQUESTED** [review]
> Reviewed 15 of 16 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617)* status: 15 of 16 files reviewed, 1 unresolved discussion (waiting on @EddeCCC)

___
*[`inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/instrumentation/hook/actions/span/ContinueOrStartSpanAction.java` line 180 at r2](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617#-Ne2oZ7L2J6lv9h2pDWY:-Ne2oZ7L2J6lv9h2pDWZ:btoacmo) ([raw file](https://github.com/inspectIT/inspectit-ocelot/blob/d75ae852816ba1f1ea58ade82fd27ac50501d6f2/inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/instrumentation/hook/actions/span/ContinueOrStartSpanAction.java#L180)):*
> ```Java
>         }
>         if (name == null) {
>             if(methodInfo != null)
> ```

these can be combined

File: inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/exporter/BrowserPropagationHttpExporterService.java [hunk 1]
```diff
@@ -107,19 +104,13 @@ protected boolean startServer(String host, int port, String path, HttpServlet se
...
+     * If the time expires, clean up the storage
      */
     @Scheduled(fixedDelay = FIXED_DELAY)
     public void updateSessionStorage() {
         if(httpServlet == null) return;
         sessionStorage.cleanUpData(timeToLive);
-
-        if(inspectitConfig == null) return;
-        int sessionLimit = inspectitConfig.getExporters().getTags().getHttp().getSessionLimit();
-        sessionStorage.setSessionLimit(sessionLimit);
```
Comment (EddeCCC) [inline inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/exporter/BrowserPropagationHttpExporterService.java:123]: Updates of exporter properties will restart the exporter. Thus, no updates have to be done in the scheduled method.

**Comment by EddeCCC** [comment]
- add createRemoteParentContext()
- Patch Tags HTTP-Exporter
- add session_id_header-property
- update dependencies

**Review by EddeCCC — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617)* status: 14 of 29 files reviewed, 2 unresolved discussions (waiting on @TitusLabs)

___
*[`inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/instrumentation/hook/actions/span/ContinueOrStartSpanAction.java` line 180 at r2](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617#-Ne2oZ7L2J6lv9h2pDWY:-NebhNDHD7QgjBB6bIci:b-896fix) ([raw file](https://github.com/inspectIT/inspectit-ocelot/blob/d75ae852816ba1f1ea58ade82fd27ac50501d6f2/inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/instrumentation/hook/actions/span/ContinueOrStartSpanAction.java#L180)):*
<details><summary><i>Previously, TitusLabs (Titus Meyer) wrote…</i></summary><blockquote>

these can be combined
</blockquote></details>

Done.

**Review by TitusLabs — CHANGES_REQUESTED** [review]
> Reviewed 13 of 15 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617)* status: all files reviewed, 2 unresolved discussions (waiting on @EddeCCC)

___
*[`inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/exporter/BrowserPropagationServlet.java` line 0 at r4](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617#-NegIhv5BHbsdhGIHU3D:-NegIhv5BHbsdhGIHU3E:bbxblbu) ([raw file](https://github.com/inspectIT/inspectit-ocelot/blob/682f94e267635de4a3b4f93edb7a46ccddd8420c/inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/exporter/BrowserPropagationServlet.java#L0)):*
Where's the check for allowed origins?

**Review by EddeCCC — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617)* status: 23 of 29 files reviewed, 2 unresolved discussions (waiting on @TitusLabs)

___
*[`inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/exporter/BrowserPropagationServlet.java` line  at r4](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617#-NegIhv5BHbsdhGIHU3D:-NehtRR67TRZLUCcSZ6I:b-896fix) ([raw file](https://github.com/inspectIT/inspectit-ocelot/blob/682f94e267635de4a3b4f93edb7a46ccddd8420c/inspectit-ocelot-core/src/main/java/rocks/inspectit/ocelot/core/exporter/BrowserPropagationServlet.java#L)):*
<details><summary><i>Previously, TitusLabs (Titus Meyer) wrote…</i></summary><blockquote>

Where's the check for allowed origins?
</blockquote></details>

Done.

**Review by TitusLabs — APPROVED** [review]
> Reviewed 10 of 10 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/inspectIT/inspectit-ocelot/1617)* status: all files reviewed, 1 unresolved discussion

**Pull request merged**